tests/run-make/libs-through-symlinks/rmake.rs

```diff
@@ -0,0 +1,32 @@
+//! Regression test for [rustc doesn't handle relative symlinks to libraries
+//! #13890](https://github.com/rust-lang/rust/issues/13890).
+//!
+//! The rust compiler searches by default for libraries in its current directory, but used to have
+//! difficulty following symlinks leading to required libraries if the real ones were located
+//! elsewhere. After this was fixed in #13903, this test checks that compilation succeeds through
```

Comment:
"elsewhere" as in, in a different directory?

Reply:
As in, to the best of my knowledge, this original test meant that

```
foo/
    libactual.rlib
bar/ # $CWD set; rustc -L . bar.rs where bar.rs depends on a foo
    libsymlink.rlib --> ../foo/libactual.rlib
```

Reply:
I tried to clarify this a bit.

Reply:
ah something like the following?

```
but used to have difficulty with symlinks to libraries located outside this subtree.
```

Reply:
reworded to:

```rs
//! This smoke test checks that for a given library search path `P`:
//!
//! - `rustc` is able to locate a library available via a symlink, where:
//!     - the symlink is under the directory subtree of `P`,
//!     - but the actual library is not (is in a different directory subtree).
//!
//! For example:
//!
//! ```text
//! actual_dir/
//!     libfoo.rlib
//! symlink_dir/  # $CWD set; rustc -L . bar.rs that depends on foo
//!     libfoo.rlib --> ../actual_dir/libfoo.rlib
//! ```
//!
//! Previously, if `rustc` was invoked with CWD set to `symlink_dir/`, it would fail to traverse the
//! symlink to locate `actual_dir/libfoo.rlib`. This was originally fixed in
//! <https://github.com/rust-lang/rust/pull/13903>.
```

Reply:
nice, looks good -- with fixed indentation for the bullet points and "it's in a different directory"

Reply:
(The indentation is intentional, the "is" was also intentional but can be a bit confusing so I'll reword that)

Reply:
Fixed.